csvGenerator.ipynb

```diff
@@ -0,0 +1,152 @@
+    "    os.makedirs(folder_path, exist_ok=True)\n",
+    "    filename = os.path.join(folder_path, f\"{file_id}.pdf\")\n",
+    "    headers = {\n",
+    "        \"Authorization\": f\"Bearer {citesphere_token}\",\n",
+    "        \"Content-Type\": \"application/pdf;charset=UTF-8\"\n",
+    "    }\n",
+    "    response = requests.get(giles_url, headers=headers)\n",
+    "    if response.status_code == 200:\n",
+    "        with open(filename, \"wb\") as file:\n",
+    "            file.write(response.content)\n",
+    "        return filename  \n",
+    "    else:\n",
+    "        return None  \n",
+    "\n",
+    "def addToCSV(itm:list,csv_dict:dict):\n",
+    "    for value in itm[\"items\"]:\n",
+    "        if value[\"key\"] in csv_dict:\n",
+    "            continue\n",
+    "        value[\"path\"] = \"\"\n",
+    "        if value[\"gilesUploads\"]:\n",
+    "            giles_id = value[\"gilesUploads\"][0][\"uploadedFile\"][\"id\"]\n",
+    "            if giles_id != \"\":\n",
+    "                value[\"path\"] = downloadFile(giles_id, auth_object.access_token)\n",
+    "        csv_dict[value[\"key\"]] = value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "229c7e47",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "PDF downloaded to: citesphere-connector/FILEQL8sKmrswS7p.pdf\n",
+      "PDF downloaded to: citesphere-connector/FILENf3i3D2v2XkG.pdf\n",
+      "PDF downloaded to: citesphere-connector/FILEcpf2PgsbX0YR.pdf\n",
+      "PDF downloaded to: citesphere-connector/FILEwUpCCo2Y5Tmx.pdf\n",
+      "Authentication failed with status code 500\n",
+      "FILEcHWoIjAxJSlV\n",
+      "PDF downloaded to: citesphere-connector/FILE3u01O8ugYrge.pdf\n",
+      "PDF downloaded to: citesphere-connector/FILESSuu8mSuZRdM.pdf\n",
+      "Authentication failed with status code 500\n",
+      "FILEfItViY7iUgLM\n"
+     ]
+    }
+   ],
+   "source": [
+    "csv_dict={}\n",
+    "for group in groups:\n",
+    "    groupID = group[\"id\"]\n",
+    "    collections = connector.get_collections(groupID)\n",
+    "    for x in collections[\"collections\"]:\n",
+    "        numPages = math.ceil(x[\"numberOfItems\"] / 50)\n",
+    "        if numPages == 0:\n",
+    "            continue\n",
+    "\n",
+    "        if numPages == 1:\n",
+    "            itm = connector.get_collection_items_pg(groupID, x[\"key\"], 1)\n",
+    "            addToCSV(itm,csv_dict)\n",
+    "        else:\n",
+    "            for i in range(numPages):\n",
+    "                itm = connector.get_collection_items_pg(groupID, x[\"key\"], i + 1)\n",
+    "                addToCSV(itm,csv_dict)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "d42eaa42",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with open('citesphereCSV.csv', 'w', newline='') as file:\n",
```

Comment:
metadata should be written to the csv as files are being downloaded

Reply:
fixed that.